src/create_venv.sh

```diff
@@ -1,136 +1,258 @@
 #!/usr/bin/env bash
 
+# Copyright (c) YugaByte, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied.  See the License for the specific language governing permissions and limitations
+# under the License.
+#
 
-# Use SHA of requirements.txt as a comment in requirements_frozen.txt
-# 
-
-function text_file_sha() {
-  local file="${1}"
-  local tmp
-  tmp="$(sort -u <<<"$(grep -v '^#' "${file}")")"
-  awk '{print $1}'<<<"$(sha256sum <<<"${tmp}")"
-}
+[[ "${_YB_CREATE_VENV_INCLUDED:=""}" == "yes" ]] && return 0
+_YB_CREATE_VENV_INCLUDED=yes
+YB_VERBOSE=${YB_VERBOSE:-false}
 
 set -e -u -o pipefail
 
-YB_PYTHON_VERSION=${YB_PYTHON_VERSION:-3.7}
-py_major_version=$(cut -f1 -d. <<<"${YB_PYTHON_VERSION}")
+_src_dir="${BASH_SOURCE%/*}"
+if [[ ! -d "${_src_dir}" ]]; then _src_dir="$PWD"; fi
```

Comment:
What case are we trying to handle? How could the directory containing the script not exist?

Reply:
When it is `.`

Reply:
Sorry, I misspoke and was thinking of something else.
The reason is because the file can be both sourced or executed.
```
jharveysmith@system76-pc ~/code/yugabyte-bash-common (new_venv*?) $ cat test.sh
#!/usr/bin/env bash

_src_dir="${BASH_SOURCE%/*}"

echo $_src_dir
jharveysmith@system76-pc ~/code/yugabyte-bash-common (new_venv*?) $ cat test2.sh
#!/usr/bin/env bash

. test.sh
jharveysmith@system76-pc ~/code/yugabyte-bash-common (new_venv*?) $ ./test2.sh
test.sh
jharveysmith@system76-pc ~/code/yugabyte-bash-common (new_venv*?) $ ./test.sh
.
```
We are checking that _src_dir is a directory and not a file, not that if it exists of doesn't exist.